internal/mode/static/state/graph/gateway_listener.go

```diff
@@ -238,8 +238,8 @@ func getAndValidateListenerSupportedKinds(listener v1.Listener) (
 
 	supportedKinds := make([]v1.RouteGroupKind, 0, len(listener.AllowedRoutes.Kinds))
 
-	validHTTPRouteKind := func(kind v1.RouteGroupKind) bool {
-		if kind.Kind != v1.Kind("HTTPRoute") {
+	validHTTPProtocolRouteKind := func(kind v1.RouteGroupKind) bool {
```

Comment:
There's a return value above that needs GRPCRoute added to it.

Reply:
I'm not sure that should have it added - I'll need to check, but I think adding that causes a conformance test to fail. Also IIRC, that return value is setting the default supported route to HTTPRoute - I don't know if we should add GRPCRoute to the default supported kinds (especially seeing as the feature is currently gated behind an experimental flag).